### PR TITLE
Classes 17 20

### PR DESCRIPTION
**Pull request by jimflores5**
Please review classes 17 -20 (TypeScript and Angular lessons 1 - 3).

### CODE REVIEW

File: Class-17-(TypeScript).md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+1. Review the distinction between *classes* and *interfaces* in TypeScript.
+1. Review the syntax for function declarations in TypeScript:
+    1. The type must be identified for each parameter.
+    1. If a value is returned, the type for that value must also be identified.
+1. Note that when a TypeScript file is compiled, the process generates a JavaScript file that can be run.
+1. Studio intro.
+
+## Studio (TF Notes)
+
+1. Remind the students that they need to be in the master branch of the TypeScript repository. Also remind them to switch to the "studio" folder to find the starter code.
```
Comment (gildedgardenia) [inline Class-17-(TypeScript).md]: I would add some clarity to this. If I had not read the actual studio instructions, I would not know that this was a commentary on how to properly fork the repo.

File: Class-18-(Angular-Lesson-1).md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    1. Pull up the LaunchCode website (launchcode.org) and use the dev tools to examine how many different parts make up the page.
+    1. What features do the different pages of the website have in common?
+    1. How do templates help us? (Class discussion).
+1. Review the Angular CLI commands for creating a new project and generating new components.
+1. Can the HTML tag defined for a nested component be used directly in the ``app.component.html`` file? Explore with the students to see.
+1. Studio intro.
+
+## Studio (TF Notes)
+
+1. If you do not know Angular well, you must ABSOLUTELY complete ALL of the prep work yourself! We expect the Angular lessons to be a big lift, and your students need you to be in top form.
```
Comment (gildedgardenia) [inline Class-18-(Angular-Lesson-1).md]: I would rephrase this: "If you do not know Angular well, you need to do the prep work! The Angular lessons are a big lift and students really need your support and technical prowess to succeed."

File: Class-19-(Angular-Lesson-2).md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+1. Discuss the relative scale/priority for the different Angular directives - components, structural directives, attribute directives.
+1. Remind the students that only one structural directive can be placed inside an HTML tag. With this in mind, ask how we would approach a situation where we want to use two of these directives?
+1. Live code examples for ``*ngFor`` and ``*ngIf``.
+    1. Discuss the difference between *hiding* HTML elements vs. *removing* them.
+1. If time permits, discuss commonly used events besides the ``keyup``, ``keydown``, ``click``, and ``mouseover`` presented in the reading.
+1. Studio intro.
+
+## Studio (TF Notes)
+
+1. If you do not know Angular well, you must ABSOLUTELY complete ALL of the prep work yourself! Be the coach your students need.
```
Comment (gildedgardenia) [inline Class-19-(Angular-Lesson-2).md]: I would rephrase this: "Remember to do the prep work if you are struggling with Angular yourself!"

File: Class-20-(Angular-Lesson-3).md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+1. Dynamic style changes.
+    1. Must combine data-binding with an event handler.
+1. Relate components to functions - they should be small, accomplish only one simple task, and be usable in many different projects.
+1. Recognize that Angular does a LOT more than what we covered in the last 3 chapters.
+1. Studio intro.
+
+## Studio (TF Notes)
+
+1. Celebrate the last studio of unit 1 with your team!
+1. If you do not know Angular well, you must ABSOLUTELY complete ALL of the prep work yourself! Seriously, if you didn't do this for the first two Angular studios, shame on you. Fix that and finish strong for your students.
```
Comment (gildedgardenia) [inline Class-20-(Angular-Lesson-3).md]: The way this point is phrased you don't need the second sentence.

File: Class-20-(Angular-Lesson-3).md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+1. Make sure your students begin in the ``studio-3`` branch of the ``mission planner`` repository.
+1. Students need to include logical && or || in order to enable/disable the cargo buttons as described in the instructions. Be ready to coach them on how to add these operators to the conditionals in the ``[disabled]`` binding.
```
Comment (gildedgardenia) [inline Class-20-(Angular-Lesson-3).md]: Missing some inline code!

**Review by gildedgardenia — CHANGES_REQUESTED** [review]
I think these look great! I only had a few minor edits.

File: Class-18-(Angular-Lesson-1).md (hunk 2 above)
Comment (jimflores5) [reply Class-18-(Angular-Lesson-1).md]: Done.  Thanks.

File: Class-17-(TypeScript).md (hunk 1 above)
Comment (jimflores5) [reply Class-17-(TypeScript).md]: Hmmm.  This was not actually a commentary about forking the repo, which just reinforces the need for clarity.

I split this into two revised bullet points.  Let me know what you think.

File: Class-19-(Angular-Lesson-2).md (hunk 3 above)
Comment (jimflores5) [reply Class-19-(Angular-Lesson-2).md]: Rephrased.  Thanks.

File: Class-20-(Angular-Lesson-3).md (hunk 4 above)
Comment (jimflores5) [reply Class-20-(Angular-Lesson-3).md]: Rephrased.  Let me know what you think.

File: Class-20-(Angular-Lesson-3).md (hunk 5 above)
Comment (jimflores5) [reply Class-20-(Angular-Lesson-3).md]: ``&&`` and ``||`` marked.

**Review by gildedgardenia — APPROVED** [review]
Approved!
